Hyper/Animation/Model.cs

```diff
@@ -26,30 +27,48 @@ public Model(string modelPath, string texturePath)
     public void Render(Shader shader, float scale, Vector3 cameraPosition, Vector3 modelPosition, Matrix4 modelRotation)
     {
         _texture.Use(TextureUnit.Texture0);
-        
+
         var modelLs = Matrix4.CreateTranslation((modelPosition - cameraPosition) * scale);
         var scaleLs = Matrix4.CreateScale(scale);
         shader.SetMatrix4("model", scaleLs * modelRotation * modelLs);
-        
+
         Animator.Animate(_model);
-        
+
         for (int i = 0; i < _model.Meshes.Count; i++)
         {
-            var boneTransforms = Animator.GetBoneTransforms(_model, i).Select(AssimpToOpenTk).ToArray();
+            var boneTransforms = Animator.GetBoneTransforms(_model, i).Select(Conversions.ToOpenTKMatrix).ToArray();
             shader.SetMatrix4Array("boneTransforms", boneTransforms);
-            
+
             GL.BindVertexArray(_vaos[i]);
             GL.DrawElements(PrimitiveType.Triangles, _model.Meshes[i].FaceCount * 3,
                 DrawElementsType.UnsignedInt, 0);
         }
     }
-    
-    private static Matrix4 AssimpToOpenTk(Matrix4x4 assimpMatrix)
+
+    public void RenderFromPose(RigidPose rigidPose, Shader shader, float globalScale, Vector3 cameraPosition, float localScale, Vector3 localTranslation)
     {
-        return new Matrix4(
-            assimpMatrix.A1, assimpMatrix.A2, assimpMatrix.A3, assimpMatrix.A4,
-            assimpMatrix.B1, assimpMatrix.B2, assimpMatrix.B3, assimpMatrix.B4,
-            assimpMatrix.C1, assimpMatrix.C2, assimpMatrix.C3, assimpMatrix.C4,
-            assimpMatrix.D1, assimpMatrix.D2, assimpMatrix.D3, assimpMatrix.D4);
+        _texture.Use(TextureUnit.Texture0);
+
+        var localTranslationMatrix = Matrix4.CreateTranslation(localTranslation);
```

Comment:
shouldn't local transform be multiplied by scale?

Reply:
That depends on whether we are going to specify local translation in respect to scaled or not scaled model. It just seemed more intuitive to me to do it like that